external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -725,6 +725,16 @@ abstract class AvroSuite
     assert(result.sameElements(expected))
   }
 
+  test("support user provided avro schema url") {
+    val avroSchemaUrl = testFile("test_sub.avsc")
+    val result = spark.read.option("avroSchemaUrl", avroSchemaUrl)
+      .format("avro")
+      .load(testAvro)
+      .collect()
+    val expected = spark.read.format("avro").load(testAvro).select("string").collect()
+    assert(result.sameElements(expected))
+  }
```

Comment:
Could you add more test cases? For example,
1. A test coverage for both `avroSchema` and `avroSchemaUrl` exist.
2. A negative test case for a wrong `avroSchemaUrl`.